src/infrastructure/database/models/user_list/search_history_database_handler.py

```diff
@@ -0,0 +1,41 @@
+from bson import ObjectId
+from datetime import datetime
+from pymongo.collection import Collection
+
+from src.infrastructure.database.models.user_list.search_history import UserSearchHistory
+
+
+class SearchHistoryHandler:
+    @staticmethod
+    async def get_user_search_history_user_id(
+            limit: int,
+            offset: int,
+            search_history_collection: Collection,
+            alcohols_collection: Collection,
+            user_id: str = None,
+    ) -> list[(dict, datetime)]:
+        search_history = list(search_history_collection.find({'user_id': ObjectId(user_id)}, {'alcohols': 1}))
+
+        search_history = search_history[0]['alcohols']
+        alcohol_ids = [a_dict['alcohol_id'] for a_dict in search_history]
```

Comment:
zrób to jednym for'em zamiast dwoma
po prostu:
for alcohol_dict in search_history:
    alcohol_ids.append(..)
    dates.append(...)